### PR TITLE
Trev8939/statistical query

### PR DESCRIPTION
**Pull request by TADraeseke**
https://devtopia.esri.com/runtime/java-android-sdk/issues/1446

Demonstrates some of the new statistical query functionality.

@shellygill - very UI heavy, so this review may take longer than others.

### CODE REVIEW

**Comment by TADraeseke** [comment]
Currently blocked on https://devtopia.esri.com/runtimecore/c_api/issues/10070

**Comment by TADraeseke** [comment]
@shellygill - This is ready to go now, the core bug was my fault (serializing something I shouldn't). Should be crash free now!

**Comment by tschie** [comment]
Instead of showing a warning when you try to add an item from the Grouped By list to the Order By list, you could just set the checkbox to active when you click on the list item.

File: java/statistical-query-group-sort/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <string name="add">Add</string>
+    <string name="type">Type:</string>
+    <string name="field">Field:</string>
+    <string name="remove_statistic">Remove Statistic</string>
+    <string name="left"><![CDATA[<<]]></string>
+    <string name="right"><![CDATA[>>]]></string>
+    <string name="group_fields">Group Field(s):</string>
+    <string name="order_by_field">Order by Field</string>
+    <string name="get_statistics">Get Statistics</string>
+    <string name="change_sort_order">Change sort order</string>
```
Comment (tschie) [inline java/statistical-query-group-sort/src/main/res/values/strings.xml]: Maybe "Reverse sort order"? I think the button should also be disabled if nothing is selected. Also, when I click it, it looks like the ascending/descending is flipping back and forth a couple times before being set.

**Comment by tschie** [comment]
Sometimes there is a delay after clicking Get Statistics before results are shown. Can we show a spinner or some progress to indicate the request was sent?

**Comment by tschie** [comment]
Get Statistics should be disabled if the statistics list is empty.

File: java/statistical-query-group-sort/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <string name="us_states_census">https://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer/3
```
[the remark this answers is not in the document]
Comment (tschie) [reply java/statistical-query-group-sort/src/main/res/values/strings.xml]: Scratch that. We decided on keeping the same dataset.

**Comment by eri9600** [comment]
@TADraeseke  The app tends to crash when clicking **CHANGE SORT ORDER** and nothing is selected 
under "Order by field", so disabling the button if nothing is selected could fix this. 
Also,  when adding more than 1 statistic, then  removing the one in the bottom, and then clicking the **REMOVE STATISTIC** the app crashes. The error being printed is an IndexOutOfBoundsException. I fixed this by adding an if statement that checks if the position is within bounds on removeStatistic() function.

**Comment by eri9600** [comment]
@TADraeseke The copy right year at the top of the files needs to be updated to 2018.

File: java/statistical-query-group-sort/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/ResultsActivity.java [hunk 3]
```diff
@@ -16,18 +16,18 @@
 
 package com.esri.arcgisruntime.sample.statisticalquery;
 
+import java.util.LinkedHashMap;
+import java.util.List;
```
Comment (eri9600) [inline java/statistical-query-group-sort/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/ResultsActivity.java]: Add an extra space

File: java/statistical-query-group-sort/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/ResultsActivity.java [hunk 4]
```diff
@@ -16,18 +16,18 @@
...
 import android.content.Intent;
```
Comment (eri9600) [inline java/statistical-query-group-sort/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/ResultsActivity.java]: Add an extra space in between java imports and android imports.

File: java/statistical-query-group-sort/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+   * Helper method to get a field from a string.
+   *
+   * @param fieldAndOrder string from recycler view
+   * @return field as a string
+   */
+  private static String getFieldFrom(String fieldAndOrder) {
+    return fieldAndOrder.substring(0, fieldAndOrder.indexOf(' '));
+  }
+
+  @Override protected void onPause() {
```
Comment (eri9600) [inline java/statistical-query-group-sort/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/MainActivity.java]: I think you also need onResume and onDestroy

Comment (TADraeseke) [reply java/statistical-query-group-sort/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/MainActivity.java]: onPause is only there to dismiss the alert. We use onResume and onDestroy to dispose of the MapView, but there isn't a mapview in this sample.

**Comment by TADraeseke** [comment]
Thanks @eri9600 - added those fixes :)

**Pull request merged**